python/src/main/resources/grpc/python/ipython_server.py

```diff
@@ -72,29 +76,30 @@ def _output_hook(msg):
         payload_reply = []
         def execute_worker():
             reply = self._kc.execute_interactive(request.code,
-                                            output_hook=_output_hook,
-                                            timeout=TIMEOUT)
+                                        output_hook=_output_hook,
+                                        timeout=TIMEOUT)
             payload_reply.append(reply)
 
         t = threading.Thread(name="ConsumerThread", target=execute_worker)
-        t.start()
-
-        while t.is_alive():
-            while not stdout_queue.empty():
-                output = stdout_queue.get()
-                yield ipython_pb2.ExecuteResponse(status=ipython_pb2.SUCCESS,
-                                                  type=ipython_pb2.TEXT,
-                                                  output=output)
-            while not stderr_queue.empty():
-                output = stderr_queue.get()
-                yield ipython_pb2.ExecuteResponse(status=ipython_pb2.ERROR,
-                                                  type=ipython_pb2.TEXT,
-                                                  output=output)
-            while not image_queue.empty():
-                output = image_queue.get()
-                yield ipython_pb2.ExecuteResponse(status=ipython_pb2.SUCCESS,
-                                                  type=ipython_pb2.IMAGE,
-                                                  output=output)
+        with self._lock:
```

Comment:
Indeed, it will. But this is the expected behavior. Right now, if you do an other request to the kernel (auto-complete), the `execute_interactive` will never complete.